File: nim/armstrong-numbers/Iteration1/armstrong_numbers.nim
import math
  
proc isArmstrongNumber*(number: int): bool =
  var digits: seq[int]
  var count: int
  var current = number

  while current != 0:
    digits.add(current mod 10)
    current = current div 10
    count = count + 1

  var total: int

  for digit in digits:
    total = total + digit ^ count

  number == total
    

File: zig/armstrong-numbers/Iteration1/armstrong_numbers.zig
const std = @import("std");

pub fn isArmstrongNumber(num: u128) bool {
    if (num == 0) return true;
    
    const digits = std.math.log10_int(num) + 1;

    var curr = num;
    var sum: u128 = 0;

    while (curr > 0) {
        const digit = curr % 10;
        sum += std.math.pow(u128, digit, digits);
        curr = curr / 10;
    }
    
    return num == sum;
}
